demo_usage/cuboids_to_bboxes.py

```diff
@@ -122,37 +121,41 @@ def dump_clipped_3d_cuboids_to_images(
         log_calib_data = dl.get_log_calibration_data(log_id)
-        ply_fpaths = dl.get_ordered_log_ply_fpaths(log_id)
+
         flag_done = False
-        for i, ply_fpath in enumerate(ply_fpaths):
-            if i % 50 == 0:
-                logging.info(f"\tOn file {i} of {log_id}")
-            lidar_timestamp = ply_fpath.split("/")[-1].split(".")[0].split("_")[-1]
-            lidar_timestamp = int(lidar_timestamp)
-            lidar_pts = load_ply(ply_fpath)
-
-            city_to_egovehicle_se3 = dl.get_city_to_egovehicle_se3(log_id, lidar_timestamp)
-            if city_to_egovehicle_se3 is None:
-                continue
-
-            labels = dl.get_labels_at_lidar_timestamp(log_id, lidar_timestamp)
-            if labels is None:
-                logger.info(f"\tLabels missing at t={lidar_timestamp}")
-                continue
-
-            for cam_idx, camera_name in enumerate(RING_CAMERA_LIST + STEREO_CAMERA_LIST):
-                camera_counter = 0
-                # load images, e.g. 'image_raw_ring_front_center_000000486.jpg'
-                im_fpath = dl.get_closest_im_fpath(log_id, camera_name, lidar_timestamp)
-                if im_fpath is None:
+        for cam_idx, camera_name in enumerate(RING_CAMERA_LIST + STEREO_CAMERA_LIST):
+            cam_im_fpaths = dl.get_ordered_log_cam_fpaths(log_id, camera_name)
+            for i, im_fpath in enumerate(cam_im_fpaths):
+                if i % 50 == 0:
+                    logging.info(f"\tOn file {i} of camera {camera_name} of {log_id}")
+
+                cam_timestamp = im_fpath.split("/")[-1].split(".")[0].split("_")[-1]
+                cam_timestamp = int(cam_timestamp)
+
+                # load PLY file path, e.g. 'PC_315978406032859416.ply'
+                ply_fpath = dl.get_closest_lidar_fpath(log_id, cam_timestamp)
+                if ply_fpath is None:
                     continue
-                save_img_fpath = f"{save_dir}/{camera_name}_{lidar_timestamp}.jpg"
+                lidar_pts = load_ply(ply_fpath)
+                save_img_fpath = f"{save_dir}/{camera_name}_{cam_timestamp}.jpg"
                 if Path(save_img_fpath).exists():
                     saved_img_fpaths += [save_img_fpath]
                     if max_num_images_to_render != -1 and len(saved_img_fpaths) > max_num_images_to_render:
                         flag_done = True
                         break
                     continue
+
+                city_to_egovehicle_se3 = dl.get_city_to_egovehicle_se3(log_id, cam_timestamp)
+                if city_to_egovehicle_se3 is None:
+                    continue
+
+                lidar_timestamp = ply_fpath.split("/")[-1].split(".")[0].split("_")[-1]
+                lidar_timestamp = int(lidar_timestamp)
+                labels = dl.get_labels_at_lidar_timestamp(log_id, lidar_timestamp)
+                if labels is None:
+                    logger.info(f"\tLabels missing at t={lidar_timestamp}")
```

Comment:
Same comment about the logging.

Reply:
Thanks, what's the advantage @mgilson-argo ?

Reply:
First and foremost, it's considered a "best-practice".  The reason that it's a best practice can be demonstrated by the following class:

```py
def Foo:
    def __str__(self) -> str:
        time.sleep(10)
        return "Well, that took a while..."
```

If you log this thing:

```py
instance = Foo()
logging.debug(f"instance: {instance}")
```

You'll take ~10s to make the log statement even if your log level is set such that you'll never get a chance to see the message.  If you defer the formatting by passing it as format strings to the logger:

```py
logging.debug("instance: %s", instance)
```

then you'll save yourself 10s each time you hit this line (unless your log level is `DEBUG` or lower).  Basically, it's a performance boost for formatting expensive strings.  Most of the time it doesn't matter, but once in a while it does.  Because of that, it's a best practice to just always defer the formatting and let the logger handle it.

(the logger will also do things like catch and re-log errors if you happen to specify the format string incorrectly whereas doing it greedily will cause exceptions -- though this is less of a problem in a world where we have `f-strings`).